analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -29,5 +29,101 @@ public class PublicMethodArgumentsShouldBeCheckedForNull : SymbolicRuleCheck
 
     protected override DiagnosticDescriptor Rule => S3900;
 
-    public override bool ShouldExecute() => false;
+    public override bool ShouldExecute()
+    {
+        return IsAccessibleFromOtherAssemblies(Node)
+            && (IsRelevantMethod(Node) || IsRelevantPropertyAccessor(Node));
+
+        static bool IsRelevantMethod(SyntaxNode node) =>
+            node is BaseMethodDeclarationSyntax { } method
+            && MethodDereferencesArguments(method);
+
+        static bool IsRelevantPropertyAccessor(SyntaxNode node) =>
+            node is AccessorDeclarationSyntax accessor
+            && IsPropertyAccessorAccessibleFromOtherAssemblies(accessor.Modifiers);
+
+        static bool IsAccessibleFromOtherAssemblies(SyntaxNode node) =>
```

Comment:
None of these 3 needs to be static with `node` argument. They all have access to `Node` and they are never called with any other value.